### ISSUE TITLE
Stacked bar chart doesn't display correct group's values on hover

### ISSUE THREAD

**Issue by Timelessprod** [metabase/metabase#44070]
### Describe the bug

I have a table with some numeric values, a category column and date column. I created a question from it using the editor (not SQL). I grouped by this table on the category and date->month with aggregation being the sum of the numeric column (nothing fancy). Note that for some of the types the values are always negative. I created a stacked bar chart in the visualisation options (x is date->month, y is sum of the numeric value per category) and reorder the groups (aka categories block). Everything looked good at first, the table is filled correctly.

However when I hover the stacks it shows me only one category value regardless of the group color I hover and sometimes the value is NaN (see screenshots).

<img width="354" alt="Screenshot 2024-06-12 at 17 10 22" src="https://github.com/metabase/metabase/assets/69579602/5e206be7-9a9e-4098-8002-20154f5d3ad7">
<img width="494" alt="Screenshot 2024-06-12 at 17 10 40" src="https://github.com/metabase/metabase/assets/69579602/fa78ff7d-a260-4d0c-9b3e-3b1b028d9628">

It's not happening to all stacked bar charts. I tried to filter out negative values but I either:
- Get the same result if I put the filter just after the data source in the editor
- Get no chart at all (aka the grey default chart usually displayed when the chart is loading, but nothing load) if I put the filter after the summarize.

### To Reproduce

1. Go to Metabase, create a question from a table which has at least a decimal column, a date/timestamp column and a category column (string). Maybe the table needs to have only negative values for some categories.
2. In the editor, summarize the table with a sum of the numeric column grouped by the category column and the date column by month.
3. Click visualize, check that the table look nice (no null values)
4. Click visualization, select "Bar".
5. In the settings reorder some of the X-axis groups and set the stacking to "Stack", set the "stacked chart type" to bars, turn on "Show values on data point", with "Values to show" set to "All" and "Auto formatting" to "Auto".
6. Click "done"
7. The bar chart looks ok but when hovering the stacks you can notice the bug.

### Expected behavior

When hovering a stack's group I should see this group's value with other groups' values and their % as shown below:
<img width="507" alt="Screenshot 2024-06-12 at 17 41 38" src="https://github.com/metabase/metabase/assets/69579602/6c846394-527c-4927-95fd-6e90bf2fc58d">

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.11+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "17.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.11+9",
    "os.name": "Linux",
    "os.version": "6.1.58+",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "bigquery-cloud-sdk"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-06-07",
      "tag": "v0.49.15",
      "hash": "55821ac"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Annoying

**Comment by JesseSDevaney** [comment]
Hey @Timelessprod!

Thank you for reporting the issue you are experiencing. I just picked this up to start working on it, but I am unable to reproduce your issue.

---
Would you be able to check if the issue is still occurring? 

And if so, provide more detailed reproduction steps like what the actual data looks like - provided that any _sensitive information_ you do not want to be seen is removed.

More specifically, if this is still occurring, would you be able to send a view of what your reproducible data looks like and note which exact stack/category is experiencing the issue?

For example:
![image](https://github.com/metabase/metabase/assets/22608765/67368f93-0c6e-4557-9c6a-ddf91a51cf1a)
"It is the 2024 year stack that has broken tool-tip values displayed"

**Comment by Timelessprod** [comment]
The bug has been resolved by upgrading to v0.50.10 :D
